PWGGA/GammaConv/AliAnalysisTaskGammaConvDalitzV1.cxx

```diff
@@ -3317,24 +3317,64 @@ Double_t AliAnalysisTaskGammaConvDalitzV1::GetPsiPair( const AliDalitzAODESD *tr
   //      Mas   ter Thesis. Thorsten Dahms. 2005
   // https://twiki.cern.ch/twiki/pub/ALICE/GammaPhysicsPublications/tdahms_thesis.pdf
   //
-  Double_t momPos[3];
-  Double_t momNeg[3];
-  if( trackPos->GetPxPyPzG(momPos) == 0 ) trackPos->GetPxPyPzG( momPos );
-  if( trackNeg->GetPxPyPzG(momNeg) == 0 ) trackNeg->GetPxPyPzG( momNeg );
-
-  TVector3 posDaughter;
-  TVector3 negDaughter;
+  Double_t momPos[3]={0.0,0.0,0.0};
+  Double_t momNeg[3]={0.0,0.0,0.0};
+  Double_t fPos[3]={0.0,0.0,0.0};
+
+        TVector3 posDaughterB;
+        TVector3 negDaughterB;
+        TVector3 posDaughterA;
+        TVector3 negDaughterA;
+  if (fAODESDEvent->GetIsESD()){
+    if( trackPos->GetPxPyPzG(momPos) == 0 ) trackPos->GetPxPyPzG( momPos );
+    if( trackNeg->GetPxPyPzG(momNeg) == 0 ) trackNeg->GetPxPyPzG( momNeg );
+        posDaughterA.SetXYZ( momPos[0], momPos[1], momPos[2] );
+        negDaughterA.SetXYZ( momNeg[0], momNeg[1], momNeg[2] );
+        posDaughterB.SetXYZ( momPos[0], momPos[1], momPos[2] );
+        negDaughterB.SetXYZ( momNeg[0], momNeg[1], momNeg[2] );
+  }
+  else {
+   // Double_t dzp[2],covdzp[3],dzn[2],covdzn[3];
+    AliExternalTrackParam Positive;
+    Positive.CopyFromVTrack(trackPos->GetDalitzVTrack());
```

Comment:
Because on this part I called with `CopyFromVTrack`, and do not let me use with `const AliDalitzAODESD`, I try a cast and did not work, so in principle, I believe that the solution easier is to remove the `const`. On the next part, I work with these tracks to calculate the propagate momentum and use the PsiPair cut.

Reply:
ok thanks, I was just curious for the reason